packages/back-end/src/models/UserModel.ts

```diff
@@ -52,6 +52,46 @@ export async function getAllUsers(): Promise<UserInterface[]> {
   return users.map((u) => toInterface(u));
 }
 
+export async function getAllUsersFiltered(
+  page: number,
+  search?: string
+): Promise<UserInterface[]> {
+  const query: {
+    $or?: [{ name: unknown }, { email: unknown }];
+  } = {};
+  if (search) {
+    query["$or"] = [
+      { name: { $regex: `${search}.*`, $options: "i" } },
+      { email: { $regex: `${search}`, $options: "i" } },
+    ];
+  }
+
+  const docs = await getCollection(COLLECTION)
+    .find(query)
+    .sort([["dateCreated", -1]])
+    .skip((page - 1) * 50)
+    .limit(50)
+    .toArray();
+
+  // return the user interface but filter out password hash
+  return docs
+    .map((u) => toInterface(u))
+    .map((u) => ({ ...u, passwordHash: "" }));
```

Comment:
It is probably better to list the properties that you actually want. If another sensitive property gets added in the future then we might forget to update this.